vroom/dartfmt.vroom

```diff
@@ -0,0 +1,68 @@
+The built-in dartfmt formatter knows how to format dart code.
+If you aren't familiar with basic codefmt usage yet, see main.vroom first.
+
+We'll set up codefmt and configure the vroom environment, then jump into some
+examples.
+
+  :source $VROOMDIR/setupvroom.vim
+
+  :let g:repeat_calls = []
+  :function FakeRepeat(...)<CR>
+  |  call add(g:repeat_calls, a:000)<CR>
+  :endfunction
+  :call maktaba#test#Override('repeat#set', 'FakeRepeat')
+
+  :call codefmt#SetWhetherToPerformIsAvailableChecksForTesting(0)
+
+
+The dartfmt formatter expects the dartfmt executable to be installed on your
+system.
+
+  % main(){}
+  :FormatCode dartfmt
+  ! dartfmt .*
+  $ main(){}
+
+The name or path of the dartfmt executable can be configured via the
+dartfmt_executable flag if the default of "dartfmt" doesn't work.
+
+  :Glaive codefmt dartfmt_executable='mydartfmt'
+  :FormatCode dartfmt
+  ! mydartfmt .*
+  $ main(){}
+  :Glaive codefmt dartfmt_executable='dartfmt'
+
+
+You can format any buffer with dartfmt specifying the formatter explicitly.
+
+  @clear
+  % main() { print("hello ");
+  |print("world\n");}
+
+  :FormatCode dartfmt
+  ! dartfmt .*2>.*
+  $ main() {
+  $ \tprint("hello ")
+  $ \tprint("world\\n")
+  $ }
+  main() {
+  	print("hello ")
+  	print("world\n")
+  }
+  @end
+
+The dart filetype will use the dartfmt formatter by default.
+
+  @clear
+  % main(){}
+
+  :set filetype=dart
+  :FormatCode
+  ! dartfmt .*
```

Comment:
I always suggest at least filing a feature request in the external project to implement context-aware range formatting. Up to you whether you want to support range formatting like this. It gives the option to people who know what they're doing, but could confuse others who don't.